### PR TITLE
fix(use-popover-positioning): minor style and offset adjustments

### LINKED ISSUE #3108: [FlyoutMenu] slightly incorrect auto positioning
### What package is this for?

- [x] React
- [ ] Angular

### Describe the bug

The menu auto positioning is off by a pixel

### To Reproduce

1. Go to https://next.carbon-addons-iot-react.com/?path=/story/1-watson-iot-flyoutmenu--auto-positioning-example
2. Click on the buttons to expand the menus

> Please create a reduced test case in CodeSandbox:
> https://codesandbox.io/s/github/carbon-design-system/carbon-addons-iot-react/tree/next/packages/react/examples/codesandbox

### Expected behavior

Pixelperfection

### Environment/versions:

- OS: MacOS
- Browser: Chrome Version 96.0.4664.55

### Additional context

image

### Specific timeline issues / requests

Do you want this work within a specific time period? Is it related to an
upcoming release?

_NB: The core contributors will try to work with your timeline, but it's not
guaranteed. The earlier you make a request in advance of a desired delivery
date, the better!_

### PR DESCRIPTION
**Pull request by kevinsperrine**
Closes #3108 

**Summary**

- fixes minor positioning issues on FlyoutMenu
- fixes minor style issues on FlyoutMenu in RTL
- adds VRT for auto-positioning of FlyoutMenu in RTL/LTR

**Acceptance Test (how to verify the PR)**

- check FlyoutMenu auto-positioning story and confirm everything is correct in RTL/LTR

**Regression Test (how to make sure this PR doesn't break old functionality)**

- check Tooltip auto-positioning story and confirm everything is correct in RTL/LTR
- check OverflowMenu auto-positioning story and confirm everything is correct in RTL/LTR

**Things to look for during review**

- [ ] Make sure all references to `iot` or `bx` class prefix is using the prefix variable
- [ ] (React) All major areas have a `data-testid` attribute. New test ids should have test written to ensure they are not changed or removed.
- [ ] UI should be checked in RTL mode to ensure the proper handling of layout and text.
- [ ] All strings should be translatable.
- [ ] The code should pass a11y scans (The storybook a11y knob should show no violations). New components should have a11y test written.
- [ ] Unit test should be written and should have a coverage of 90% or higher in all areas.
- [ ] All components should be passing visual regression test. For new styles or components either a visual regression test should be written for all permutations or the base image updated.
- [ ] Changes or new components should either write new or update existing documentation.
- [ ] PR should link and close out an existing issue

### CODE REVIEW

File: packages/react/src/components/FlyoutMenu/FlyoutMenu.test.e2e.jsx [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+          display: 'flex',
+          alignItems: 'flex-start',
+          justifyContent: 'flex-start',
+          flex: 1,
+        }}
+      >
+        <FlyoutMenu
+          direction={FlyoutMenuDirection.LeftEnd}
+          useAutoPositioning
+          menuOffset={{ top: 2, left: isRTL ? -2 : 0 }}
```
Comment (kevinsperrine) [inline packages/react/src/components/FlyoutMenu/FlyoutMenu.test.e2e.jsx:30]: I've had to manually adjust the positioning of these with menuOffset objects. They are off by a few pixel in Cypress, but look perfect in storybook. I don't know if these offsets will need to be adjusted over time, or if I should leave them off and take the snapshots with them in the "wrong" position since they are correct in storybook/chrome.

File: packages/react/src/hooks/usePopoverPositioning.test.jsx [hunk 2]
```diff
@@ -88,7 +88,7 @@ describe('usePopoverPositioning', () => {
     userEvent.click(button);
     const menu = screen.getByTestId(testId);
 
-    expect(menu).toHaveClass(`${iotPrefix}--flyout-menu--body__${FlyoutMenuDirection.LeftEnd}`);
```
Comment (kevinsperrine) [inline packages/react/src/hooks/usePopoverPositioning.test.jsx:91]: This test was incorrect and is changed, because of the flyoutAlignment changes that fixes an edge case above: https://github.com/carbon-design-system/carbon-addons-iot-react/pull/3133/files#diff-c56a39e4778901918956ade07124779426ef2c20782d0ac1bdc7d7981805c08cR205

File: packages/react/src/components/FlyoutMenu/FlyoutMenu.test.e2e.jsx (hunk 1 above)
Comment (bjornalm) [reply packages/react/src/components/FlyoutMenu/FlyoutMenu.test.e2e.jsx:30]: I also tested in the Storybook iframe canvas where it looked good, it is strange that they are off here. If you leave the adjustments here we should add a comment like the above to the code that explains why.

**Review by bjornalm — APPROVED** [review]
LGTM. It would be better if possible to find the root cause of the required adjustments, but I assume you have investigated that?

File: packages/react/src/components/FlyoutMenu/FlyoutMenu.test.e2e.jsx (hunk 1 above)
Comment (kevinsperrine) [reply packages/react/src/components/FlyoutMenu/FlyoutMenu.test.e2e.jsx:30]: Yeah, I spent quite-a-while yesterday trying different padding on the cypress window, different sizes, etc. I don't know. I'll copy my comment into the code, though.

Comment (davidicus) [reply packages/react/src/components/FlyoutMenu/FlyoutMenu.test.e2e.jsx:36]: That is aggravating!

Comment (davidicus) [reply packages/react/src/components/FlyoutMenu/FlyoutMenu.test.e2e.jsx:36]: I wonder if these offsets would be necessary if we used grid instead of flexbox (just an example). Wondering if its just a weird bug from Cypress's implementation

**Review by davidicus — APPROVED** [review]
I think this should be fine. I wonder if it makes more sense to leave out the offsets but keep in a note of the discrepancy in case we need to investigate in the future.

**Comment by kevinsperrine** [comment]
> 

Yeah, You can make an argument for it either way. I don't really know which is best.
